src/blueprint/azext_blueprint/custom.py

```diff
@@ -0,0 +1,577 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=too-many-statements
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-locals
+# pylint: disable=unused-argument
+
+import json
+import os
+from knack.util import CLIError
+
+
+def import_blueprint_with_artifacts(cmd,
+                                    client,
+                                    blueprint_name,
+                                    input_path,
+                                    management_group=None,
+                                    scope=None):
+    from ._client_factory import cf_artifacts
+
+    artifact_client = cf_artifacts(cmd.cli_ctx)
+    body = {}
+    blueprint_path = os.path.abspath(os.path.join(input_path, 'blueprint.json'))
+
+    with open(blueprint_path) as blueprint_file:
+        blueprint = json.load(blueprint_file)
+        if blueprint['properties'] is None:
+            raise CLIError('Blueprint file does not contain properties field')
+        blueprint_properties = blueprint['properties']
+        if 'displayName' in blueprint_properties:
+            body['display_name'] = blueprint_properties['displayName']  # str
+        if 'description' in blueprint_properties:
+            body['description'] = blueprint_properties['description']  # str
+        if 'targetScope' in blueprint_properties:

```

Comment:
Not a big deal, but this if else could be made into a one liner: body['target_scope'] = blueprint_properties.get('targetScope', 'subscription')

Reply:
updated